ui/apps/platform/src/hooks/useFetchDeploymentCount.ts

```diff
@@ -19,12 +23,15 @@ const DEPLOYMENT_COUNT_QUERY = gql`
     }
 `;
 
-function useFetchDeploymentCount(searchFilter: SearchFilter): UseFetchDeploymentCount {
+function useFetchDeploymentCount(
+    searchFilter: SearchFilter,
+    queryOptions: QueryHookOptions<DeploymentCountResponse, DeploymentCountParameters> = {}
+): UseFetchDeploymentCount {
     const query = getRequestQueryStringForSearchFilter(searchFilter);
-    const queryOptions = { variables: { query } };
+    const options = { ...queryOptions, variables: { query } };
```

Comment:
Only question is whether `query` is the only property in `variables` object?

That is, this spread update idiom is shallow replacement, not deep merge (which might be needed to update only one among various properties at second level of data hierarchy.

Reply:
Good point, the intention here was to allow access to `useQuery`'s options via the caller, but without the ability to override the variables. I'll make this more apparent with
```ts
Omit<
        QueryHookOptions<DeploymentCountResponse, DeploymentCountParameters>,
        'variables'
>
```